oai_agents/agents/base_agent.py

```diff
@@ -440,19 +440,12 @@ def evaluate(self, eval_agent, num_eps_per_layout_per_tm=5, visualize=False, tim
         return np.mean(tot_mean_reward), rew_per_layout
 
 
-    def set_new_teammates(self):
+    def set_new_teammates(self, curriculum=None):
         for i in range(self.args.n_envs):
             layout_name = self.env.env_method('get_layout_name', indices=i)[0]
-            
-            pop_teamtypes = self.teammates_collection[layout_name]
-            population = [pop_teamtypes[t] for t in pop_teamtypes.keys()]
-            
-            '''
-            population = [[[agent1, agent2, agent3], ...], [[agent3, agent5, agent6], ...], ...]
-            '''
+            population_teamtypes = self.teammates_collection[layout_name]
 
-            teammates_per_type = population[np.random.randint(len(population))]
-            teammates = teammates_per_type[np.random.randint(len(teammates_per_type))]
+            teammates = self.curriculum.select_teammates(population_teamtypes=population_teamtypes)
```

Comment:
I think this should be  
```
teammates = curriculum.select_teammates(...)
```
Because later in `rl.py` you pass in `self.curriculum` as the `curriculum` argument

Reply:
Thanks for catching that!

Reply:
I added a new commit that addresses all your comments!